### DISCUSSION TITLE
Problem! use stm32 to driver st7735

### DISCUSSION THREAD

**Discussion by shuiyihang** [zephyrproject-rtos/zephyr#34628] [Q&A]
hi,everybody
I encountered a problem，When I use spi and dma to drive st7735 screen with a size of 129*160, I found that its refreshing speed is slower than that of pure hardware spi. What happened? Which direction should I try to solve the problem?
the board is stm32F102RCT6
the config is this :
`&spi1 {
	pinctrl-0 = <&spi1_nss_master_pa4 &spi1_sck_master_pa5
		     &spi1_miso_master_pa6 &spi1_mosi_master_pa7>;
	status = "okay";
	dmas = <&dma1 3 1 0x20440 0x00>,
            <&dma1 2 1 0x20480 0x00>;
    dma-names = "tx", "rx";

	cs-gpios = <&gpioa 4 GPIO_ACTIVE_LOW>;
	lcd_st7735:st7735@0{
			compatible = "sitronix,st7735";
			reg=<0>;
			spi-max-frequency = <18000000>; /* 8MHz 36000000*/
			label = "ST7735";
			reset-gpios = <&gpiob 12 GPIO_ACTIVE_HIGH>;
			blk_gpios = <&gpioa 1 GPIO_ACTIVE_HIGH>;
			cmd-data-gpios = <&gpioc 4 GPIO_ACTIVE_HIGH>;
			width = <129>;
			height = <160>;
	};
};`

**Comment by erwango** [comment]
We may have an issue on DMA support on F1.
Can you have a try on https://github.com/zephyrproject-rtos/zephyr/pull/34666 ?

**Reply by shuiyihang** [reply]
hi ,I merged your branch and used st,stm32-dma-v2bis, but I found that data->dma_tx.dma_dev is always empty in the function spi_stm32_init, even though I did all the configuration I thought it should be, I just went straight Two sentences are added above the judged to be empty, namely: data->dma_tx.dma_dev=device_get_binding("DMA_1");
data->dma_rx.dma_dev=device_get_binding("DMA_1");
Finally my st7735 screen can be used, but the speed seems to be slower! !

**Reply by shuiyihang** [reply]
Can you tell me how to use logging? I have enabled logging in the configuration item and set the filtering level, but when I use log_inf, there is still no feedback from the serial port, and I can only see printk prints. Output

**Reply by erwango** [reply]
Can you comment directly in the PR, it is currently in review, so it's possible it is buggy.

**Reply by shuiyihang** [reply]
yes，I have done this now
